accounts-db/src/accounts_db.rs

```diff
@@ -3913,7 +3875,7 @@ impl AccountsDb {
     }
 
     /// Shrinks `store` by rewriting the alive accounts to a new storage
-    fn shrink_storage(&self, store: &AccountStorageEntry) {
+    fn shrink_storage(&self, store: Arc<AccountStorageEntry>) {
```

Comment:
Thanks for the comments. I think we are fine to use `Arc` here. There are two callers for this function. Both of them move the `Arc` at the call. There are no clone happening. `&Arc` doesn't help much here.

Reply:
I think even if you passed the &Arc, you still have to `defer` and `clone` on line 3909. 

The store is borrowed in 3898 inside `shrink_collect` and used in 3927.

Reply:
You have to clone once, either here or in shrink_collect...

Reply:
> Thanks for the comments. I think we are fine to use Arc here. There are two callers for this function. Both of them move the Arc at the call. There are no clone happening.

The two callers are not cloning. They are moving the Arc at the call site.

Reply:
Arc is like *p.  &Arc is like **p

let's say we want to read member A in *p. We just need to load from (*p+offset).
But to read A in **p, you need do two loads. one load to get the address of *p. Then load it again from (*p+offset).

With &Arc, we are paying one extra load in the memory.

Reply:
Also, when future code added to use storage after shrink call, rust compiler will fail to compile because of borrow checker. By then, we can consider to use &Arc.